src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -534,23 +572,69 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
         }
         match(
             operation, [](...) { llvm_unreachable("NOT YET IMPLEMENTED"); },
-            [&](AALoad)
+            [&](OneOf<AALoad, BALoad, CALoad, DALoad, FALoad, IALoad, LALoad, SALoad>)
             {
+                llvm::Type* type;
+
+                match(
```

Comment:
Match statements support `return` so you can write:
```cpp
type = match(operation, ...);
```
you just have to watch out that all lambdas return a compatible return type (for the `[](...)` you'll likely need to add `-> llvm::Type*` explicitly)